fission/function.go

```diff
@@ -497,3 +528,36 @@ func fnPods(c *cli.Context) error {
 
 	return err
 }
+
+func fnTest(c *cli.Context) error {
+	fnName := c.String("name")
+	routerURL := os.Getenv("FISSION_ROUTER")
+	if len(routerURL) == 0 {
+		fatal("Need FISSION_ROUTER set to your fission router.")
+	}
+
+	url := fmt.Sprintf("http://%s/fission-function/%s", routerURL, fnName)
+
+	fnReachable := false
+
+	resp := httpRequest(c.String("method"), url, c.String("body"), c.StringSlice("header"))
+	if resp.StatusCode < 300 {
+		body, err := ioutil.ReadAll(resp.Body)
+		checkErr(err, "Function test")
+		fmt.Print(string(body))
+		defer resp.Body.Close()
+		fnReachable = true
+	}
+
+	if !fnReachable {
+		body, err := ioutil.ReadAll(resp.Body)
+		checkErr(err, "read log response from pod")
+		fmt.Printf("Error calling function %v: %v %v", fnName, resp.StatusCode, string(body))
+		defer resp.Body.Close()
+		err = printPodLogs(c)
```

Comment:
This logic -- calling print pod logs and falling back to the DB -- would be useful in the `fission fn logs` command as well.  (But we can push that to a later change if you want to.)

Reply:
@soamvasani I can integrate this and make it so that ``` fn logs``` calls pod logs and if not found then looks up DB logs. But there are two things which are inconsistencies from user perspective and we should address in future: 1) The logs from POD don't have a timestamp before every line while ```fn logs``` prints TS 2) The ```fn logs``` shows lot many lines as opposed to pod logs which shows only recent pod's logs. So the behavior/experience can vary based on if you test a function and call logs or call it without testing it. There might be some optimizations in fetching logs from DB side too. What do you think?